avm/ptn/authorization/pim-role-assignment/modules/management-group.bicep

```diff
@@ -0,0 +1,161 @@
+
+@description('Required. The type of the PIM role assignment whether its active or eligible.')
+param pimRoleAssignmentType pimRoleAssignmentTypeType
+
+@sys.description('Optional. The justification for the role eligibility.')
+param justification string = ''
+
+@sys.description('Required. The type of the role assignment eligibility request.')
+param requestType requestTypeType
+
+@sys.description('Optional. The resultant role eligibility assignment id or the role eligibility assignment id being updated.')
+param targetRoleEligibilityScheduleId string = ''
+
+@sys.description('Optional. The role eligibility assignment instance id being updated.')
+param targetRoleEligibilityScheduleInstanceId string = ''
+
+@sys.description('Optional. The resultant role assignment schedule id or the role assignment schedule id being updated.')
+param targetRoleAssignmentScheduleId string = ''
+
+@sys.description('Optional. The role assignment schedule instance id being updated.')
+param targetRoleAssignmentScheduleInstanceId string = ''
+
+@sys.description('Optional. Ticket Info of the role eligibility.')
+param ticketInfo ticketInfoType?
+
+@sys.description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to.')
+param condition string = ''
+
+@sys.description('Optional. Version of the condition. Currently accepted value is "2.0".')
+@allowed([
+  '2.0'
+])
+param conditionVersion string = '2.0'
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Resource Policy Contributor': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '36243c78-bf99-498c-9df9-86d9f8d28608'
+  )
+  'Role Based Access Control Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f58310d9-a9f6-439a-9e8d-f62e7b41a168'
+  )
+  'User Access Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9'
+  )
+  'Management Group Reader': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'ac63b705-f282-497d-ac71-919bf39d939d'
+  )
+}
+
+var roleDefinitionIdVar = (builtInRoleNames[?roleDefinitionIdOrName] ?? roleDefinitionIdOrName)
+
+resource pimEligibleRoleAssignment 'Microsoft.Authorization/roleEligibilityScheduleRequests@2022-04-01-preview' = if (pimRoleAssignmentType.roleAssignmentType == 'Eligible') {
+  name: guid(deployment().name, managementGroupId, roleDefinitionIdVar, principalId)
+  properties: {
+    principalId: principalId
+    roleDefinitionId: roleDefinitionIdVar
+    requestType: requestType
+    condition: !empty(condition) ? condition : null
```

Comment:
As opposed to the parent module, here you actually pass in `null` if nothing was provided.

To simplify your code, I'd suggest to replace all condition parameters with 
```bicep
param condition string?
```
and remove all checks. So in this case it would just be
```bicep
    condition: condition
``` 

This will lead to the behavior: If the user passes anything in, it will be passed into the deployment, otherwise `null` is used.

Reply:
resolved